Kauri support

funkemunky opened the pull request.

JustinDevB commented:
GitHub was behind so now there are a lot of conflicts.

I added Kauri Support, and tweaked a few things with your ideas. Also did not want to fully incorporate Atlas and force users to have to install Atlas if they are not running Kauri

funkemunky commented:
> GitHub was behind so now there are a lot of conflicts.
> 
> 
> 
> I added Kauri Support, and tweaked a few things with your ideas. Also did not want to fully incorporate Atlas and force users to have to install Atlas if they are not running Kauri

You would not have to have forced users to use Atlas. The way I implemented the code would allow it to work without Atlas
